### DISCUSSION TITLE
The "interpolated snippet" in LuaSnip.

### DISCUSSION THREAD

**Discussion by no-vici** [L3MON4D3/LuaSnip#882] [Q&A]
Hi LuaSnip enthusiasts,

I had these two snippets a while ago, the former works as expected for its simplicity, and for some arbitrary reasons I moved to the latter one and struggled to interpret its meaning. I asked ChatGPT and I got this [answer](https://zzzcode.ai/answer-question?id=956d9051-2b62-4dde-90a2-fea8f1b58bdb). 

1. `s("do", fmt("do\n\t{1}\nend", i(1, "--do_body")))`
2. ` s("do", { isn(1, t({ "do", "" }), i(1, "--do_body"), "") })`

The logics from the AI's answer sound reasonable, the 2nd snippet worked before but now I got the error bellow. Could someone kindly explain to me what are the advantages of the second snippet over the first one and what the error means please?

![image](https://user-images.githubusercontent.com/113570767/236727308-afbc08ff-0e65-481f-be90-5d11d701299e.png)

**Comment by L3MON4D3** [comment]
Whew, maybe our jobs are safe for a while yet :P
`isn` is an indentSnippetNode, it works like a snippetNode, but accepts an additional parameter which dictates the indentation of the nodes (look for it in `DOC.md` for more details) (but it seems unnecessary in this case.?)

In any case, it should work if the nodes passed to the isn are wrapped in a list

**Reply by no-vici** [reply]
> `isn` is an indentSnippetNode.

I would blame the AI for this ;) I've had a looked at the `DOC.md` for a while, forgotten most of the concepts.

> In any case, it should work if the nodes passed to the isn are wrapped in a list.

The syntax looks right, right? I will come back to the original version as it supports indented blocks nicely, it did not before, and that was why I looked up for the second version.

Thank you for looking into this.

Cheers.

**Reply by L3MON4D3** [reply]
You're welcome :)

Quickly, why 2 has no indenttion: the thing with `isn` is that all indentation, even the indentation of the trigger is disregarded (this is so that text can be inserted at column 0, regardless of the triggers indentation (probably useful for stuff like `#ifdef` in c/c++)), and it can be reinstated by passing `"$PARENT_INDENT"` to the `isn`.
But version 1 looks perfectly fine, version two would only be more elegant if there were additional lines in the do-end-block

**Reply by no-vici** [reply]
> ... and it can be reinstated by passing `"$PARENT_INDENT"` to the isn.

Oh, I see, didn't think about the `"$PARENT_INDENT"`. I will try this.

Thank you once more.

**Reply by no-vici** [reply]
> ... it can be reinstated by passing `"$PARENT_INDENT"` to the `isn`.

Wow, that does the trick. Thank you, thank you.

`s("do", { isn(1, t({ "$PARENT_INDENT" .. "do", "$PARENT_INDENT" }), i(1, "$PARENT_INDENT" .. "--do_body"), "$PARENT_INDENT" .. "" ) })`

**Reply by L3MON4D3** [reply]
Anytime :)
